### PR TITLE
CIP-1694 | Bootstrap & other fixes

### PR DESCRIPTION
**Pull request by WhatisRT**
This mostly aligns the description of the Bootstrap phase with what we're actually going to do. Additionally, it fixes some other small things that people have noticed.

### CODE REVIEW

File: CIP-1694/README.md
Comment (Ryun1) [inline CIP-1694/README.md:1111]: "Info actions will be available, but no other actions are possible during the bootstrap phase."

maybe this is confusing phrasing, as both hardforks and protocol parameter change governance actions are also supported.

**Review by rphair — APPROVED** [review]
Approving pending change @Ryun1 recommended in https://github.com/cardano-foundation/CIPs/pull/800#discussion_r1577761175 which seems completely straightforward (and maybe also https://github.com/cardano-foundation/CIPs/pull/800#discussion_r1577777704), so @WhatisRT I've put this on the agenda for & tagged as `Last Check` since I think this PR can go through as a matter of course, if not already merged before next week's meeting: https://hackmd.io/@cip-editors/87

File: CIP-1694/README.md
Comment (rphair) [reply CIP-1694/README.md]: resolving: fixed in https://github.com/cardano-foundation/CIPs/pull/800/commits/9ab965eea626d4fd091e898d6f4e21aabdd02c63

File: CIP-1694/README.md
Comment (rphair) [reply CIP-1694/README.md]: resolving: fixed in https://github.com/cardano-foundation/CIPs/pull/800/commits/9ab965eea626d4fd091e898d6f4e21aabdd02c63

**Comment by Ryun1** [comment]
A few more nit picks from me;
- [Line 420](https://github.com/cardano-foundation/CIPs/blob/a59635744500530c37037fc660db733bb98f4166/CIP-1694/README.md?plain=1#L420C106-L420C110) "associated stake credential is also delegated to a DRep." -> "associated stake credential is also delegated to a DRep or pre-defined voting option."
- [Line 426](https://github.com/cardano-foundation/CIPs/blob/a59635744500530c37037fc660db733bb98f4166/CIP-1694/README.md?plain=1#L426C19-L426C35) "(including to a pre-defined DRep)" -> "(including to a pre-defined voting options)"
- [Line 334](https://github.com/cardano-foundation/CIPs/blob/a59635744500530c37037fc660db733bb98f4166/CIP-1694/README.md?plain=1#L334C129-L334C165) "by voting on any governance actions." -> "by voting on any governance actions or submitting a DRep update certificate."
- [Line 347](https://github.com/cardano-foundation/CIPs/blob/a59635744500530c37037fc660db733bb98f4166/CIP-1694/README.md?plain=1#L347C1-L347C62) "The following new types of certificates will be added for DReps: DRep registration certificates, DRep retirement certificates, and vote delegation certificates." -> also needs DRep update certs
- [Line 725](https://github.com/cardano-foundation/CIPs/blob/a59635744500530c37037fc660db733bb98f4166/CIP-1694/README.md?plain=1#L725) "Votes can be cast multiple times for each governance action by a single credential." -> "by a voting credential."
- [Line 1034](https://github.com/cardano-foundation/CIPs/blob/a59635744500530c37037fc660db733bb98f4166/CIP-1694/README.md?plain=1#L1034C1-L1034C59) "We propose to implement the changes via **one hard fork**." -> We propose to implement the changes via **two hard forks**.

**Comment by thenic95** [comment]
Here are my comments before the Rationale:

- **Row 186**: The current text hints at implementing a guardrails script during the bootstrap phase. Considering we are definitively implementing a smart-contract-based guardrail for the interim constitution, we should revise the text to clearly reflect this update.
- **Row 206**: It would be beneficial to explicitly mention the guardrail script here, along with the minimum and maximum values for the Proposal Policy (PP). This detail is crucial because any value outside these defined limits should be considered "unconstitutional."
- **Row 245**: Should we specify the parameter name `committeeMinSize`? Including the precise term can enhance clarity and ensure consistency across our documentation.
- **Row 262**: Instead of expressing the threshold for the Constitutional Committee (CC) as a fraction, it should be noted as a percentage or a decimal number, aligning with the format presented in [Conway Genesis Settings](https://book.world.dev.cardano.org/environments/sanchonet/conway-genesis.json). This adjustment will improve readability and consistency in our numerical representations.
- **Row 457** : Governance Action `3. Update to the Constitution or proposal policy` . I believe it is a grave mistake for combining the Constitution and the proposals policy/guardrails in one governance action. The Cardano Constitution holds a higher value in my opinion than a proposal policy that just exists to reduce the possibilities of human error of an inexperience Constitutional Committee. Changing the Constitution should be among the most difficult parameters to change, while adapting a proposal policy is something that cannot be linked to the constitution. What if in future node versions or other improvements of the Cardano network currently selected upper and lower bounds of this proposal policy become obsolote and we need to change them as the circumstances change. Then we would need to change the entire constitution. I see this as an unnecessary hassle as the core values and guiding principles are not effected, but rather the protocol boundaries. I would advocate for disconnecting the proposal policy and the constitution.
- **Row 576**: In the current implementation a GA is enacted one epoch after it has been ratified. The CIP should explicitly mention this. This could also be mentioned in the chapter `Lifecycle`
- **Row 691**: Should we include the initial recommendation for the governance parameters once the Parameter Committee provides them? Should be after the next meeting

**Comment by WhatisRT** [comment]
I've addressed everything so far, except the following points:

> * [Line 347](https://github.com/cardano-foundation/CIPs/blob/a59635744500530c37037fc660db733bb98f4166/CIP-1694/README.md?plain=1#L347C1-L347C62) "The following new types of certificates will be added for DReps: DRep registration certificates, DRep retirement certificates, and vote delegation certificates." -> also needs DRep update certs

This depends on your viewpoint, the spec only adds one certificate. From my POV the split into registration and update is an implementation detail.

> * [Line 725](https://github.com/cardano-foundation/CIPs/blob/a59635744500530c37037fc660db733bb98f4166/CIP-1694/README.md?plain=1#L725) "Votes can be cast multiple times for each governance action by a single credential." -> "by a voting credential."

There's no such thing as a voting credential. Any credential can vote (in principle), so this sentence isn't wrong. In practice, most credentials will probably exist for the sole purpose of casing votes, but not all.

> * **Row 186**: The current text hints at implementing a guardrails script during the bootstrap phase. Considering we are definitively implementing a smart-contract-based guardrail for the interim constitution, we should revise the text to clearly reflect this update.

In the CIP what people call the 'Constitution Script' or 'Guardrails Script' is referred to as the proposal policy. This is a more aspirational (probably unrealistic) version that would remove the constitution text altogether.

> * **Row 206**: It would be beneficial to explicitly mention the guardrail script here, along with the minimum and maximum values for the Proposal Policy (PP). This detail is crucial because any value outside these defined limits should be considered "unconstitutional."

The point of this example is that while we can have some hard conditions that can be enforced, the CC may have to make decisions about softer rules like that. The proposal policy or the guardrails can't really cover that. Though that specific example may be outdated, so I could replace it with something more clearly outside of the scope of the guardrails.

> * **Row 457** : Governance Action `3. Update to the Constitution or proposal policy` . I believe it is a grave mistake for combining the Constitution and the proposals policy/guardrails in one governance action. The Cardano Constitution holds a higher value in my opinion than a proposal policy that just exists to reduce the possibilities of human error of an inexperience Constitutional Committee. Changing the Constitution should be among the most difficult parameters to change, while adapting a proposal policy is something that cannot be linked to the constitution. What if in future node versions or other improvements of the Cardano network currently selected upper and lower bounds of this proposal policy become obsolote and we need to change them as the circumstances change. Then we would need to change the entire constitution. I see this as an unnecessary hassle as the core values and guiding principles are not effected, but rather the protocol boundaries. I would advocate for disconnecting the proposal policy and the constitution.

Irregardless of whether people want that or not, it's probably unrealistic to change something like that at this point in time since it might delay the HF. Note also that the constitution doesn't need to be changed with that GA. People can simply look at the action, and if the hash for the constitution was changed they can just reject it.

> * **Row 576**: In the current implementation a GA is enacted one epoch after it has been ratified. The CIP should explicitly mention this. This could also be mentioned in the chapter `Lifecycle`

It is mentioned in `Lifecycle`: All governance actions are enacted on the epoch boundary after their ratification.

> * **Row 691**: Should we include the initial recommendation for the governance parameters once the Parameter Committee provides them? Should be after the next meeting

I'm not sure. These settings are effectively temporary, so I'm not sure they belong here. @rphair, is there any guidelines for this type of content?

**Comment by rphair** [comment]
> - (@thenic95) Row 691: Should we include the initial recommendation for the governance parameters once the Parameter Committee provides them? Should be after the next meeting
>
> (@WhatisRT https://github.com/cardano-foundation/CIPs/pull/800#issuecomment-2090451718) I'm not sure. These settings are effectively temporary, so I'm not sure they belong here. @rphair, is there any guidelines for this type of content?

The general recommendation for CIPs so far has been that these types of content should be avoided when possible:
- social factors (unless they directly affect, or provide, the Motivation)
- policies of particular companies 
- temporary conditions: especially those unlikely to persist beyond the editing process & activation of a CIP.

These rules are not coded anywhere; just my understanding based on 3 years of observance at CIP meetings & seeing these types of material challenged frequently in the GitHub editing process.  @Ryun1 could provide more insight about whether Parameter Committee interaction would be considered "company policy" or a reflection of technical / community consensus.  (cc @KtorZ)

**Comment by rphair** [comment]
@WhatisRT we ran out of time with all the `Last Check`s at the CIP meeting today but @Ryun1 @Crypto2099 I would say this is another good candidate to merge between-meetings if it's true that all the points above really have been addressed... it seems that say to me after:
1) https://github.com/cardano-foundation/CIPs/pull/800/commits/552c5f292d1a28a9a440b64dc60819fefc3131c9
2) the reservations in https://github.com/cardano-foundation/CIPs/pull/800#issuecomment-2090451718
3) my confirmation (if OK with other editors) of the final reservation in https://github.com/cardano-foundation/CIPs/pull/800#issuecomment-2091185186

No hurry but I think it would be timely to merge this as soon as @Ryun1 can review it again to ensure merging produces this produces an excellent final draft (with no follow-up PR to fix these little issues later).

**Review by Ryun1 — APPROVED** [review]

**Pull request merged**
